tests/solvers/quadratic/gw_test.py

```diff
@@ -199,22 +199,36 @@ def reg_gw(a: jnp.ndarray, b: jnp.ndarray,
   @pytest.mark.parametrize("unbalanced", [False, True])
   def test_gw_pointcloud(self, unbalanced: bool):
     """Test basic computations pointclouds."""
-
-    def reg_gw(
-        x: jnp.ndarray, y: jnp.ndarray, a: jnp.ndarray, b: jnp.ndarray
-    ) -> float:
-      geom_x = pointcloud.PointCloud(x)
-      geom_y = pointcloud.PointCloud(y)
-      tau_a, tau_b = (self.tau_a, self.tau_b) if unbalanced else (1.0, 1.0)
-      prob = quadratic_problem.QuadraticProblem(
-          geom_x, geom_y, a=a, b=b, tau_a=tau_a, tau_b=tau_b
-      )
-      solver = gromov_wasserstein.GromovWasserstein(
-          epsilon=1.0, max_iterations=10
+    geom_x = pointcloud.PointCloud(self.x)
+    geom_y = pointcloud.PointCloud(self.y)
+    tau_a, tau_b = (self.tau_a, self.tau_b) if unbalanced else (1.0, 1.0)
+    prob = quadratic_problem.QuadraticProblem(
+        geom_x, geom_y, a=self.a, b=self.b, tau_a=tau_a, tau_b=tau_b
+    )
+    solver_ent = gromov_wasserstein.GromovWasserstein(
+        epsilon=1.0, max_iterations=10
+    )
+    solvers = [
```

Comment:
Rather than having `solvers = [...]`, parametrize the test as
```python
@pytest.mark.parametrize("unbalanced,rank", [(True, -1), (False, -1), (False, 3)])
```

Reply:
good idea